validator/rpc/wallet.go

```diff
@@ -43,6 +43,14 @@ func (s *Server) HasWallet(ctx context.Context, _ *ptypes.Empty) (*pb.HasWalletR
 // CreateWallet via an API request, allowing a user to save a new
 // derived, direct, or remote wallet.
 func (s *Server) CreateWallet(ctx context.Context, req *pb.CreateWalletRequest) (*pb.WalletResponse, error) {
+	// Currently defaultWalletPath is used as the wallet directory and req's WalletPath is ignored for simplicity
+	exists, err := wallet.Exists(defaultWalletPath)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not check for existing wallet: %v", err)
+	}
+	if exists {
+		return nil, status.Error(codes.AlreadyExists, "A wallet already exists at this location.")
+	}
```

Comment:
This is part of the PR fix.  It disallows using this function to bypass currently present logic that forbids creating wallets where one already exists.